internal/cli/atlas/deployments/start.go

```diff
@@ -0,0 +1,174 @@
+		return err
+	}
+}
+
+func (opts *StartOpts) Run(ctx context.Context) error {
+	if err := opts.validateAndPrompt(ctx); err != nil {
+		return err
+	}
+
+	if strings.ToLower(opts.DeploymentType) == options.LocalCluster {
+		return opts.RunLocal(ctx)
+	}
+
+	return opts.RunAtlas()
+}
+
+func (opts *StartOpts) RunLocal(ctx context.Context) error {
+	localDeployments, err := opts.GetLocalDeploymentsWithContainers(ctx)
+	if err != nil {
+		return err
+	}
+
+	if deployment, ok := localDeployments[opts.DeploymentName]; ok {
+		if err = opts.startContainer(ctx, deployment); err != nil {
+			return err
+		}
+
+		return opts.Print(
+			admin.AdvancedClusterDescription{
+				Name: &opts.DeploymentName,
+			})
+	}
+
+	return options.ErrDeploymentNotFound
+}
+
+func (opts *StartOpts) startContainer(ctx context.Context, deployment options.Deployment) error {
+	if deployment.StateName == options.StoppedState {
+		if _, err := opts.PodmanClient.StartContainers(ctx, deployment.MongoDContainer.ID, deployment.MongoTContainer.ID); err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	if deployment.StateName == options.PausedState {
+		if _, err := opts.PodmanClient.UnpauseContainers(ctx, deployment.MongoDContainer.ID, deployment.MongoTContainer.ID); err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	if deployment.StateName == options.IdleState || deployment.StateName == options.RestartingState {
+		return nil
+	}
+
+	return ErrDeploymentIsDeleting
+}
+
+func (opts *StartOpts) RunAtlas() error {
+	r, err := opts.store.StartCluster(opts.ConfigProjectID(), opts.DeploymentName)
+	if err != nil {
+		return err
+	}
+
+	return opts.Print(r)
+}
+
+func (opts *StartOpts) validateAndPrompt(ctx context.Context) error {
+	if opts.DeploymentName == "" {
+		if err := opts.DeploymentOpts.Select(ctx); err != nil {
+			return err
+		}
+	}
+
+	if opts.DeploymentType == "" {
+		if err := opts.PromptDeploymentType("What would you like to start?"); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func StartBuilder() *cobra.Command {
+	opts := &StartOpts{}
+	cmd := &cobra.Command{
+		Use:     "start <deploymentName>",
+		Short:   "Start a deployment",
+		Args:    require.MaximumNArgs(1),
+		GroupID: "all",
+		Annotations: map[string]string{
+			"deploymentNameDesc": "Name of the deployment.",
+			"output":             startTemplate,
+		},
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			opts.config = config.Default()
+			opts.CredStore = config.Default()
+			log.SetWriter(cmd.OutOrStdout())
+
+			if err := opts.PreRunE(
+				opts.initStore(cmd.Context()),
+				opts.InitOutput(log.Writer(), startTemplate)); err != nil {
+				return err
+			}
+
+			opts.PodmanClient = podman.NewClient(log.IsDebugLevel(), log.Writer())
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 1 {
+				opts.DeploymentName = args[0]
+			}
+			return opts.Run(cmd.Context())
+		},
+	}
```

Comment:
should we add a post run here to throw logs when not logggedin/no podman? https://github.com/mongodb/mongodb-atlas-cli/blob/71cc26dff3936ceb0ca1adf4555aaa90835ba089/internal/cli/atlas/deployments/list.go#L170

Reply:
updated